azharmu3@ug140:~/ece496$ python aws_instance.py
2015-03-26 04:01:47,462 boto [DEBUG]:Using access key provided by client.
2015-03-26 04:01:47,462 boto [DEBUG]:Using secret key provided by client.
2015-03-26 04:01:47,464 boto [DEBUG]:Method: POST
2015-03-26 04:01:47,464 boto [DEBUG]:Path: /
2015-03-26 04:01:47,464 boto [DEBUG]:Data:
2015-03-26 04:01:47,464 boto [DEBUG]:Headers: {}
2015-03-26 04:01:47,464 boto [DEBUG]:Host: ec2.us-east-1.amazonaws.com
2015-03-26 04:01:47,464 boto [DEBUG]:Port: 443
2015-03-26 04:01:47,465 boto [DEBUG]:Params: {'Action': 'CreateKeyPair', 'KeyName': 'ece496_key_pair', 'Version': '2014-10-01'}
2015-03-26 04:01:47,465 boto [DEBUG]:establishing HTTPS connection: host=ec2.us-east-1.amazonaws.com, kwargs={'port': 443, 'timeout': 70}
2015-03-26 04:01:47,465 boto [DEBUG]:Token: None
2015-03-26 04:01:47,465 boto [DEBUG]:CanonicalRequest:
POST
/

host:ec2.us-east-1.amazonaws.com
x-amz-date:20150326T080147Z

host;x-amz-date
d0f8eedf23ae2ead01ccc02e0bee5a6424ac5c67c52cddccb930de4bf484ef65
2015-03-26 04:01:47,466 boto [DEBUG]:StringToSign:
AWS4-HMAC-SHA256
20150326T080147Z
20150326/us-east-1/ec2/aws4_request
c18e366714ae5a66cb5fa3ba77571993f4ad82e1cb9595e5b541293738404c70
2015-03-26 04:01:47,466 boto [DEBUG]:Signature:
1b0f53f2fcbbcea48ba0d1b32317349e678b2aa4ca7b544ba537a031390ec4f9
2015-03-26 04:01:47,466 boto [DEBUG]:Final headers: {'Content-Length': '63', 'User-Agent': 'Boto/2.36.0 Python/2.7.5 Linux/3.2.0-4-amd64', 'Host': u'ec2.us-east-1.amazonaws.com', 'X-Amz-Date': '20150326T080147Z', 'Content-Type': 'application/x-www-form-urlencoded; charset=UTF-8', 'Authorization': u'AWS4-HMAC-SHA256 Credential=AKIAJBUWTF6VVAK6KJKA/20150326/us-east-1/ec2/aws4_request,SignedHeaders=host;x-amz-date,Signature=1b0f53f2fcbbcea48ba0d1b32317349e678b2aa4ca7b544ba537a031390ec4f9'}
2015-03-26 04:01:47,569 boto [DEBUG]:wrapping ssl socket; CA certificate file=/homes/a/azharmu3/.local/lib/python2.7/site-packages/boto/cacerts/cacerts.txt
2015-03-26 04:01:47,636 boto [DEBUG]:validating server certificate: hostname=ec2.us-east-1.amazonaws.com, certificate hosts=['us-east-1.ec2.amazonaws.com', 'ec2.amazonaws.com', 'ec2.us-east-1.amazonaws.com']
2015-03-26 04:01:47,676 boto [DEBUG]:Response headers: [('transfer-encoding', 'chunked'), ('date', 'Thu, 26 Mar 2015 08:01:46 GMT'), ('server', 'AmazonEC2')]
2015-03-26 04:01:47,676 boto [DEBUG]:<?xml version="1.0" encoding="UTF-8"?>
<Response><Errors><Error><Code>UnauthorizedOperation</Code><Message>You are not authorized to perform this operation.</Message></Error></Errors><RequestID>0b484771-f8a1-43d8-b0fb-1f51a25cd467</RequestID></Response>
2015-03-26 04:01:47,677 boto [ERROR]:403 Forbidden
2015-03-26 04:01:47,677 boto [ERROR]:<?xml version="1.0" encoding="UTF-8"?>
<Response><Errors><Error><Code>UnauthorizedOperation</Code><Message>You are not authorized to perform this operation.</Message></Error></Errors><RequestID>0b484771-f8a1-43d8-b0fb-1f51a25cd467</RequestID></Response>
Traceback (most recent call last):
  File "aws_instance.py", line 13, in <module>
    KeyPair = ec2Conn.create_key_pair("ece496_key_pair")
  File "/homes/a/azharmu3/.local/lib/python2.7/site-packages/boto/ec2/connection.py", line 2866, in create_key_pair
    return self.get_object('CreateKeyPair', params, KeyPair, verb='POST')
  File "/homes/a/azharmu3/.local/lib/python2.7/site-packages/boto/connection.py", line 1207, in get_object
    raise self.ResponseError(response.status, response.reason, body)
boto.exception.EC2ResponseError: EC2ResponseError: 403 Forbidden
<?xml version="1.0" encoding="UTF-8"?>
<Response><Errors><Error><Code>UnauthorizedOperation</Code><Message>You are not authorized to perform this operation.</Message></Error></Errors><RequestID>0b484771-f8a1-43d8-b0fb-1f51a25cd467</RequestID></Response>